From: Tobias Stoeckmann <tobias@stoeckmann.org>
Date: Thu, 11 Sep 2025 19:25:25 +0200
Subject: gutils: Handle huge lines in load_user_special_dirs

If a line is longer than G_INTMAX, still treat it correctly without
possibly provoking an out of boundary read.

Origin: upstream, 2.86.1, commit:300722b6c42639891bc37cab0de0e03fb2a9e127
---
 glib/gutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index 6ff9c79..3102b07 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2293,7 +2293,7 @@ load_user_special_dirs (void)
     {
       gchar *buffer = lines[i];
       gchar *d, *p;
-      gint len;
+      size_t len;
       gboolean is_relative = FALSE;
       GUserDirectory directory;
 
